docs/secure-connections/sauce-connect/advanced/sc-kgp.md

```diff
@@ -0,0 +1,46 @@
+---
+id: sc-kgp
+title: KGP
+sidebar_label: Tunneling Protocol
+---
+import useBaseUrl from '@docusaurus/useBaseUrl';
+import Tabs from '@theme/Tabs';
+import TabItem from '@theme/TabItem';
+
+## Overview
+
+This document provides information about KGP - the tunneling protocol used by Sauce Connect Proxy&trade;
+to assist customer network and security engineering teams to better understand Sauce Connect Proxy&trade; networking.
+It's recommended to become familiar with [Sauce Connect Proxy&trade; architecture](/secure-connections/sauce-connect/advanced/sc-architecture)
+first.
+
+## KGP Tunneling Protocol
+
+Sauce Connect Proxy&trade; establishes a reverse tunnel over encrypted connection between
+[Sauce Connect Proxy&trade; client](/secure-connections/sauce-connect/advanced/sc-architecture/#sauce-connect-proxy-client) and
+[Sauce Connect Proxy&trade; server](/secure-connections/sauce-connect/advanced/sc-architecture/#sauce-connect-proxy-server).
+The reverse tunneling part is achieved via protocol called KGP.
+
+KGP is an application layer protocol that carries all HTTP(s) traffic as its payload.
+It is developed and maintained by Sauce Labs, and is used to multiplex established connections for multiple HTTP requests/responses.
```

Comment:
I want to make sure I understand the difference between Sauce Connect (on its own) and Sauce Connect + KGP (am I close):

- **Sauce Connect (on its own)**: lets you start up an SSH tunnel connection between the app/site/SC client on your local environment and the Sauce Labs server.
- **Sauce Connect + KGP**:  lets you start up a _reverse_ TLS tunnel that encrypts data sent between the app/site/SC client on your local environment and the Sauce Labs server. (does reverse mean the data is going in the opposite direction? i.e., from Sauce to the customer?)

Reply:
pretty close :)

so, 
**Sauce Connect**:  lets you start up a reverse TLS tunnel that encrypts data sent between the app/site/SC client on your local environment and the Sauce Labs server.

There is no "SSH tunnel connection", only "TLS connection that is used to establish a reverse tunnel".

Reverse is because the connection is established from a customer to Sauce Labs but requests flow are from Sauce Labs to customer.